packages/compass-aggregations/src/utils/pipeline-storage.spec.js

```diff
@@ -1,35 +1,28 @@
-/* eslint-disable no-sync */
 import fs from 'fs';
 import os from 'os';
 import path from 'path';
 import { expect } from 'chai';
 
 import { PipelineStorage } from './pipeline-storage';
 
-const initialAggregationsPath =
-  process.env.MONGODB_COMPASS_AGGREGATIONS_TEST_BASE_PATH;
-
 const createPipeline = (tmpDir, data) => {
   fs.writeFileSync(
     path.join(tmpDir, 'SavedPipelines', `${data.id}.json`),
     JSON.stringify(data)
   );
 };
 
-const pipelineStorage = new PipelineStorage();
-
 describe('PipelineStorage', function () {
   let tmpDir;
+  let pipelineStorage;
   beforeEach(function () {
-    tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'SavedPipelines'));
+    tmpDir = fs.mkdtempSync(os.tmpdir());
     fs.mkdirSync(path.join(tmpDir, 'SavedPipelines'));
```

Comment:
I would expect ensuring the path exists being done by the storages

Reply:
I was thinking of doing all the directory setup as the app starts in the main process somewhere [here](https://github.com/mongodb-js/compass/blob/main/packages/compass/src/main/application.ts#L60). That way we will ensure we have all the necessary directories setup when we try to read from them.

Reply:
nice idea! but similar to the other comment, it seems that storages have the info on what's the folder: https://github.com/mongodb-js/compass/pull/4674/files/e7050dd7b6e504b4840c150e6cde7b52d0a4d1b1#diff-cda704877ab6cd60ae17d533f713bd373be3a3458105773b5a06e290f25e997fR68

So the storage would still be the "information expert" here. By requiring `_init` to run `mkdir` we would need to give the `_init` code a piece of knowledge that storages already have, and introduce a stronger coupling: where we want to save files and how (on a filesystem vs somewhere else). 

We could consider adding a `.setup()` method instead and call that in the init code .. that way we would have a fail-fast type of experience where if is not possible to write on a directory .. compass would crash??

Or, to get rid of the coupling entirely .. just use a lazy initialization, and have mkdir being called once if it wasn't before. I would prefer this way if that's what is happening right now, so we don't incur in the risk of having new issues that we didn't have before.